build.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# rmm build script
+
+# This script is used to build the component(s) in this repo from
+# source, and can be called with various options to customize the
+# build as needed (see the help output for details)
+
+# Abort script on first error
+set -e
+
+NUMARGS=$#
+ARGS=$*
+
+# NOTE: ensure all dir changes are relative to the location of this
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean librmm rmm -v -g -n -h"
+HELP="$0 [clean] [librmm] [rmm] [-v] [-g] [-n] [-h]
+   clean  - remove all existing build artifacts and configuration (start over)
+   librmm - build and install the librmm C++ code
+   rmm    - build and install the rmm Python package
+   -v     - verbose build mode
+   -g     - build for debug
+   -n     - no install step
+   -h     - print this text
+
+   default action (no args) is to build and install 'librmm and rmm' targets
+"
+LIBRMM_BUILD_DIR=${REPODIR}/build
+RMM_BUILD_DIR=${REPODIR}/python/build
+BUILD_DIRS="${LIBRMM_BUILD_DIR} ${RMM_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+PYTHON=${PYTHON:=python}
+RAN_CMAKE=0
+
+# Set defaults for vars that may not have been defined externally
+#  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
+#         CONDA_PREFIX, but there is no fallback from there!
+INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}
+PARALLEL_LEVEL=${PARALLEL_LEVEL:=""}
+
+function hasArg {
+    (( NUMARGS != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
+}
+
+# Runs cmake if it has not been run already. Bash work directory
+# is always build directory after calling this function
+function runCMake {
```

Comment:
This is minor, but after thinking about this more the function name makes it sound like it will always run cmake. Maybe rename the function to `ensureCMakeRan` or something similar?

Reply:
Fixed. Pushed new patch. Waiting for ci tests to run (tested locally before pushing, works local)